### PR TITLE
Remove hard-coded default analysis level strings and instead use properties defined in Microsoft.NET.Sdk.Analyzers.targets

### PR DESCRIPTION
**Pull request by mavasani**

### CODE REVIEW

**Comment by mavasani** [comment]
@Youssef1313

**Review by Evangelink — APPROVED** [review]

**Review by Youssef1313 — COMMENTED** [review]
Won't this break users who consume NetAnalyzers via NuGet on old .NET SDK versions that don't have these properties defined?

File: src/Tools/GenerateDocumentationAndConfigFiles/Program.cs [hunk 1]
```diff
@@ -1522,10 +1522,9 @@ static string GetGlobalAnalyzerConfigTargetContents(string packageName, string?
...
-      <{effectiveAnalysisLevelPropName} Condition=""'$({analysisLevelPropName})' == 'none' or '$({analysisLevelPrefixPropName})' == 'none'"">4.0</{effectiveAnalysisLevelPropName}>
-      <{effectiveAnalysisLevelPropName} Condition=""'$({analysisLevelPropName})' == 'latest' or '$({analysisLevelPrefixPropName})' == 'latest'"">6.0</{effectiveAnalysisLevelPropName}>
-      <{effectiveAnalysisLevelPropName} Condition=""'$({analysisLevelPropName})' == 'preview' or '$({analysisLevelPrefixPropName})' == 'preview'"">7.0</{effectiveAnalysisLevelPropName}>
+      <{effectiveAnalysisLevelPropName} Condition=""'$({analysisLevelPropName})' == 'none' or '$({analysisLevelPrefixPropName})' == 'none'"">$(_NoneAnalysisLevel)</{effectiveAnalysisLevelPropName}>
+      <{effectiveAnalysisLevelPropName} Condition=""'$({analysisLevelPropName})' == 'latest' or '$({analysisLevelPrefixPropName})' == 'latest'"">$(_LatestAnalysisLevel)</{effectiveAnalysisLevelPropName}>
+      <{effectiveAnalysisLevelPropName} Condition=""'$({analysisLevelPropName})' == 'preview' or '$({analysisLevelPrefixPropName})' == 'preview'"">$(_PreviewAnalysisLevel)</{effectiveAnalysisLevelPropName}>
```
Comment (Youssef1313) [inline src/Tools/GenerateDocumentationAndConfigFiles/Program.cs:1527]: Note: SDK already defines those:

https://github.com/dotnet/sdk/blob/97b23dac46b6f4c3aa2816ddd7bb89e1f8b4402a/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.Analyzers.targets#L48-L50

So I think these are useful here only when consumed via NuGet on *older* SDK versions

**Comment by mavasani** [comment]
> Won't this break users who consume NetAnalyzers via NuGet on old .NET SDK versions that don't have these properties defined?

Probably, but this is the behavior the users would have truly desired. I think it is reasonable, especially given not many folks likely use the category based analysis level properties.

File: src/Tools/GenerateDocumentationAndConfigFiles/Program.cs (hunk 1 above)
Comment (Youssef1313) [reply src/Tools/GenerateDocumentationAndConfigFiles/Program.cs:1527]: I see it's not exactly a duplicate since category name could be appended.

**Pull request merged**